### ISSUE TITLE
Using a POST to perform a read operation

### ISSUE THREAD

**Issue by pparavicini** [typicode/json-server#1422]
Is passing a complex object in the request body to do a READ (like a query object to do a search) supported in json-server?

**Comment by MadhuSaini22** [comment]
No, JSON-Server does not support using a POST request to perform a read operation like passing a complex object in the request body to mimic an advanced database search. JSON-Server is designed to follow RESTful principles, where POST is used for creating resources, and GET is used for reading or retrieving data. Advanced querying should be done using GET requests with appropriate query parameters.

**Comment by selamsc** [comment]
> Is passing a complex object in the request body of a POST to do a READ (like passing a query object to mock an advanced db search) supported in json-server?

You can create a middleware to update req.method to GET, it will return your value in db.json
